### ISSUE TITLE
New mod: Random hitobject positions

### ISSUE THREAD

**Issue by McKay42** [McKay42/McOsu#64]

**Comment by rohilthopu** [comment]
Do you have any ideas on how you're going to implement this? For example, how would you take into account streams? As randomized stream note positions would be a bit ridiculous haha. For example, you could include some sort of placement algo that is based on the original position of the note and make it so that it can be a max radius away but anywhere along the circumference of that radius. Also, are you planning on making any documentation for your classes? They're somewhat unintelligible to an outsider right now haha

**Comment by McKay42** [comment]
Yes, it will be based on the time difference to the previous hitobject. I've tried completely random positions which of course make it impossible. If you want to try it right now with 1/8th random circle offsets and 1/3rd random slider offsets, redownload Alpha 24 and set osu_mod_random 1 before loading a diff.

I'll write some documentation soon. Is it really unintelligible? :/
Osu.h/Osu.cpp is the main entry point for the application, and it just creates all the other interfaces and update()s & draw()s them, it also handles keyboard & mouse input delegation.

**Comment by rohilthopu** [comment]
Sweet. I'll try that out! 

Also, I was just being a bit over-dramatic :P It's not actually unintelligible by any means but it can be kinda hard to follow sometimes. I think that's just because theres a lot going on and I haven't spent enough time in C++.

**Comment by McKay42** [comment]
closing this for now, it's not that important to keep the issue open for randomness improvements
